tests/cases/compiler/functionParameterArityMismatch.ts

```diff
@@ -12,4 +12,4 @@ f2(1);
 f2(1, 2, 3);
 f2(1, 2, 3, 4, 5);
 f2(1, 2, 3, 4, 5, 6, 7);
-f2(...[1], 2, 3, 4, 5, 6);
+f2(1, 2, 3, 4, 5, ...[6, 7]);

```

Comment:
I changed this since the previous version became OK with this PR - spreading a fixed array expression into a call is now treated just as if positional arguments would be provided (thanks to `getEffectiveCallArguments` creating synthetic expressions)